web/static/webcomponents/pages/ApiKeys/ApiKeys-home/ApiKeys-home.js

```diff
@@ -0,0 +1,107 @@
+import { fromHTML } from "../../../utils.js";
+
+const apikeysHomeTemplate = document.createElement("template");
+apikeysHomeTemplate.innerHTML = `
+<link rel="stylesheet" type="text/css" href="/static/reset.css">
+<link rel="stylesheet" type="text/css" href="/static/webcomponents/pages/apikeys/apikeys-home/apikeys-home.css">
+`;
+
+class ApiKeysHome extends HTMLElement {
+  constructor() {
+    super();
+    this.shadow = this.attachShadow({ mode: "open" });
+    this.shadow.append(apikeysHomeTemplate.content.cloneNode(true));
+    this.apiKeysData = [];
+  }
+
+  async fetchApiKeys() {
+    try {
+      const response = await fetch("/api/v1/apikeys");
+      if (!response.ok) {
+        throw new Error(`HTTP error! Status: ${response.status}`);
+      }
+      const data = await response.json();
+      this.apiKeysData = data.data.apiKeys.reverse(); // Reverse the order of the API keys
+      this.updateApiKeysComponent();
+    } catch (error) {
+      console.error("Error fetching API keys:", error);
+    }
+  }
+
+  async deleteApiKey(apiKey) {
+    try {
+      const response = await fetch(`/api/v1/apikey/${apiKey}`, {
+        method: "DELETE",
+        headers: { "Content-Type": "application/json" },
+      });
+      if (!response.ok) {
+        throw new Error(`Failed to delete API key: ${response.statusText}`);
+      }
+      this.apiKeysData = this.apiKeysData.filter((key) => key.key !== apiKey);
+      this.updateApiKeysComponent();
+      console.log(`API key deleted: ${apiKey}`); // Log deletion
+    } catch (error) {
+      console.error("Error deleting API key:", error);
+    }
+  }
+
+  async createApiKey() {
+    try {
+      const response = await fetch("/api/v1/apikey", {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+      });
+      if (!response.ok) {
+        throw new Error(`Failed to create API key: ${response.statusText}`);
+      }
+      const data = await response.json();
+      const newApiKey = { key: data.data.apiKey };
```

Comment:
Once you flatten the structure above, I think you can reduce the nesting here to go along with that fix.

Reply:
Change this to data.data.apiKeys => data.data